src/app/shared/dovetail/dovetail-audio.ts

```diff
@@ -168,7 +168,9 @@ export class DovetailAudio extends ExtendableAudio {
     this._dovetailLoading = true;
     let promise = this.currentPromise = this.dovetailFetcher.fetch(url).then(
       result => {
-        if (this.currentPromise === promise) {
+        if (!result.request.placements.length) {
+          throw 'No placements in request; falling back to raw audio';
```

Comment:
Probably shouldn't throw a string in node - can cause problems in error handling.  I'd just `new Error()` that line.